code/tools/ShortCodeRelationFinder.php

```diff
@@ -39,18 +39,22 @@ public function findPageCount($number)
     }
 
     /**
+     * @param int $number
      * @return DataList
      */
     public function getList($number)
     {
+        $number = (int) $number;
         $list = DataList::create('SiteTree');
         $where = array();
         $fields = $this->getShortCodeFields('SiteTree');
         $shortcode = DMS::inst()->getShortcodeHandlerKey();
         foreach ($fields as $ancClass => $ancFields) {
             foreach ($ancFields as $ancFieldName => $ancFieldSpec) {
+                $ancClass = Convert::raw2sql($ancClass);
```

Comment:
You call raw2sql on this same string once for each $ancFields... it'll end up over-encoding this string many times.

Ways to prevent this:
 - Never re-use a variable for safe / unsafe versions of the same string.
 - Only secure a variable immediately after retrieval instead of later on down the code

Reply:
Ha, good point